drift_dev/lib/src/analysis/resolver/file_analysis.dart

```diff
@@ -73,6 +73,14 @@ class FileAnalyzer {
               .whereType<DriftElement>()
               .followedBy(availableByDefault)
               .transitiveClosureUnderReferences()
+              .where((e){
+                if (element is DriftDatabase) {
```

Comment:
Why would this only apply to databases here?

Reply:
My understanding of what's going in the overall code is likely to blame here.

I needed to grab the instance of DriftDatabase so that I could access its `id.libraryUri` for comparison against each element that I wanted to filter out.
Maybe this file is entirely the wrong location for the logic?

Reply:
This might be flow typing not applying in nested functions, but since this is all wrapped in an `if (element is BaseDriftAccessor)` you can safely use `element!.id` instead.

My comment was that we may have a similar problem with DAOs, as those are not `DriftDatabase`s.

Reply:
Thanks - new comment at the end which encapsulates all my learnings.